### ISSUE TITLE
Parent element gets shortened to child elements for miRNA lift-over

### ISSUE THREAD

**Issue by seb-mueller** [agshumate/Liftoff#61]
I'm trying to lift-over miRNAs from miRbase, which seems to work well apart from one aspect: The primary transcript is shortend to the length of the mature miRNA.

For example, one entry in the `miRNAs.gff` file (for the old assembly), looks like this:

```
SL2.50ch01 	 . 	 miRNA_primary_transcript 	 832340 	 832581 	 . 	 + 	 . 	 ID=MI0020257
SL2.50ch01 	 . 	 miRNA                    	 832540 	 832561 	 . 	 + 	 . 	 ID=MIMAT0023610;Parent=MI0020257
```
The second line is the mature miRNA which is a subsection of the primary transcript (line 1). I've indicated this with the 'Parent' tag.

Running liftoff 1.5.2 like this:

```
liftoff -g miRNAs.gff -f file_types.txt -o new_assembly_miRNAs.gff $new $old 
```

Gives me this result (`new_assembly_miRNAs.gff`):

```
SL3.0ch01 	 Liftoff 	 miRNA_primary_transcript 	 832126 	 832147 	 . 	 + 	 . 	 ID=MI0020257;coverage=1.0;sequence_ID=1.0;extra_copy_number=0;copy_num_ID=MI0020257_0
SL3.0ch01 	 Liftoff 	 miRNA                    	 832126 	 832147 	 . 	 + 	 . 	 ID=MIMAT0023610;Parent=MI0020257;extra_copy_number=0
```

Note, that the coordinates of both lines are exactly the same, but I expect the first line to be about the same length as line 1 of the `miRNAs.gff`.
I tried to fiddle with the parameters, but no luck.
So is there a way to retain the length of the parent if its children are shorter?

Note: the content of `file_types.txt` is below:

```
miRNA
miRNA_primary_transcript
```

**Comment by agshumate** [comment]
Hi Seb,
currently no, liftoff uses the boundaries of all the children to determine the parent boundaries which works well for exons -> transcripts -> genes, but i see here with this example why that may need to change. There are a couple work arounds I can suggest. One is to remove the parent tag on the miRNA, then each feature will be handled separately. If you notice cases where the miRNA is being lifted to a location that is not a subset of the primary transcript, the other option is to create some arbitrary feature that is the parent of both the primary transcript and the mature miRNA. For example something like this 

`
SL2.50ch01 	 . 	 miRNA_gene 	 832340 	 832581 	 . 	 + 	 . 	 ID=MI0020257_gene
SL2.50ch01 	 . 	 miRNA_primary_transcript 	 832340 	 832581 	 . 	 + 	 .ID=MI0020257;Parent=MI0020257_gene
SL2.50ch01 	 . 	 miRNA                    	 832540 	 832561 	 . 	 + 	 . 	 ID=MIMAT0023610;Parent=MI0020257_gene`

**Comment by seb-mueller** [comment]
Dear Alaina, 
Thanks so much for your quick response, I already suspected something like this might be the case. Hopefully this wouldn't take too much effort to implement. Certainly it would be useful in cases like that.

As for your first workaround, when only trying to lift mature miRNAs as a seperate unit, i.e. with the following minimal `miRNAs.gff`:

```
SL2.50ch01	.	miRNA	832540	832561	.	+	.	ID=MIMAT0023610
```
It actually gives an empty output file, but `unmapped_features.txt` contains `MIMAT0023610`. This is also true for all other mature miRNA, but not the primary transcripts. 
I suspect this has to do with the size being to short (21nt). 
After crude experimenting with sizes (artificially enlarging region in gff), it appears 46nt length is required for getting results back. 
Not sure what the default parameters are and how to change them, but is this expected behaviour (this might be a new issue..)?

I'll be exploring option 2 as well, thanks for the suggestion.

**Comment by agshumate** [comment]
yes this is certainly related to the length. most likely, the sequence is too short for minimap2 to align it. you can confirm this by looking at the .sam file in the intermediate_files directory to see if this the sequence is unaligned. You may be able to get alignments by lowering the -k and -w parameters of minimap2. to pass in parameters specific to minimap2, use the -mm2_options parameter in liftoff

**Comment by seb-mueller** [comment]
After experimenting a little bit, I found option 1 not to work, since I can't get minimap2 to map short (21nt) sequences alone (I went down to -k 5 without any reported mapping for the mature miRNA). That's not to say that it won't work, but possibly minimap wasn't designed that way, also not having the "Parent" information probably leads to multipmapping issues etc.

Option 2 seems to work though, however this needs fiddeling with the gff file which is not ideal, but it works.
Fyi, A quick and hacky in vim is to copy all primary transcripts to the bottom `:g/primary/t$`, visual select the new lines, and replace ID (`:'<,'>s/ID/Parent/`) and feature-name (`:'<,'>s/miRNA_primary_transcript/dummy`) (latter for easy filter them afterwards). No need to reorder. After running, one can the delete the dummy lines in the output gff (`:g/dummy/d`).

Should I still leave the issue open since this is only a workaround?
Anyway, thanks a lot for the quick help, much appreciated!

**Comment by agshumate** [comment]
yes lets leave the issue open and i will provide a solution for it soon! glad you were able to use the work around although it is far from ideal.
